CRM/Contribute/Form/UpdateBilling.php

```diff
@@ -335,17 +335,15 @@ public function postProcess() {
 
       $activityParams = array(
         'source_contact_id' => $this->_subscriptionDetails->contact_id,
-        'activity_type_id' => CRM_Core_OptionGroup::getValue('activity_type',
-          'Update Recurring Contribution Billing Details',
-          'name'
+        'activity_type_id' => CRM_Core_PseudoConstant::getKey(
+          'CRM_Activity_DAO_Activity',
+          'activity_type_id',
+          'Update Recurring Contribution Billing Details'
         ),
         'subject' => ts('Recurring Contribution Billing Details Updated'),
         'details' => $message,
         'activity_date_time' => date('YmdHis'),
-        'status_id' => CRM_Core_OptionGroup::getValue('activity_status',
-          'Completed',
-          'name'
-        ),
+        'status_id' => CRM_Core_PseudoConstant::getKey('CRM_Activity_DAO_Activity', 'status_id', 'Completed'),
```

Comment:
We really should be using BAO activity consistently - rather than DAO CRM_Core_PseudoConstant::getKey('CRM_Activity_BAO_Activity', 'status_id', 'Completed'),

(if you fix this you could do that fixes as a separate PR because I can quick-merge them & pull out of here)